src/codeu/chat/util/Time.java

```diff
@@ -66,4 +66,10 @@ public String toString() {
 
   public static Time now() { return Time.fromMs(System.currentTimeMillis()); }
 
+  // Difference of end and start times.
+  // Ratio is the power of ten from seconds that the duration will be given in
+  // (e.i. ratio = -3 is milliseconds)
+  public static long duration(Time start, Time end, int ratio) {
+    return Math.round(Math.pow(10, -(3 + ratio)) * ((end.inMs() - start.inMs())));
```

Comment:
This seems kind of cluttered making it hard to understand. Just some suggestions: Split the math into different lines to make it more readable and/or replace numbers 3 and 10 with constant variables with meaningful names so it's easier to understand the purpose of the numbers. Also, I'm not sure if the ratio parameter is necessary if the user can't change it in any way. I would just make -3 as a constant variable so that it's easier for you /or groupmates to change if we decide we want to see more/less of the decimal values.